linera-chain/src/manager.rs

```diff
@@ -103,21 +116,41 @@ pub struct MultiOwnerManager {
     pub round_timeout: Timestamp,
 }
 
-impl MultiOwnerManager {
-    pub fn new(
-        public_keys: impl IntoIterator<Item = (Owner, (PublicKey, u64))>,
-        multi_leader_rounds: RoundNumber,
-        seed: u64,
+doc_scalar!(
+    ChainManager,
+    "The state of the certification process for a chain's next block"
+);
+
+impl ChainManager {
+    pub fn reset(
+        &mut self,
+        ownership: &ChainOwnership,
+        height: BlockHeight,
         now: Timestamp,
-    ) -> Result<Self, ChainError> {
-        let public_keys: BTreeMap<Owner, (PublicKey, u64)> = public_keys.into_iter().collect();
-        let weights = public_keys.values().map(|(_, weight)| *weight).collect();
-        let distribution = WeightedAliasIndex::new(weights)?;
+    ) -> Result<(), ChainError> {
+        *self = ChainManager::new(ownership.clone(), height.0, now)?;
+        Ok(())
+    }
+
+    pub fn is_active(&self) -> bool {
+        self.ownership.is_active()
+    }
+
+    fn new(ownership: ChainOwnership, seed: u64, now: Timestamp) -> Result<Self, ChainError> {
+        let distribution = if !ownership.owners.is_empty() {
```

Comment:
FWIW I wrote some (inefficient) code for this once here:  https://github.com/novifinancial/librabft_simulator/blob/cc2ec64d8028f2540e70ca9f9e7891c4ba72113f/bft-lib/src/configuration.rs#L65